style-guide/order-list.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+
+"""ordered list check."""
+
+from proselint.tools import memoize, existence_check
+
+@memoize
+def check_orderlist(text):
+    """Use unordered list if order of steps does not matter."""
+    err = "style-guide.order-list"
+    msg = "Use unordered list if order of steps does not matter."
+    regex = "([0-9]*\.\s[(\x00-\x09)|(\x0B-\x7F)]*\n)+"
```

Comment:
I'm curious about your regex here.

Reply:
An ordered list will be of format: ``1. some characters\n``
``[0-9]*`` checks any number of digits from 0-9, then a literal ``.`` and ``\s`` checks a space.
In the character set which is checked then: ``[(\x00-\x09)|(\x0B-\x7F)]*`` checks all characters except a newline. If a newline is also matched in this character set, any pargaraph or statement which is after a list will also be matched. So, I used this character set and then checked a ``\n``.